cas-server-support-jpa-ticket-registry/src/main/java/org/jasig/cas/ticket/registry/JpaTicketRegistry.java

```diff
@@ -121,7 +125,8 @@ private Ticket getRawTicket(final String ticketId) {
             if (ticketId.startsWith(TicketGrantingTicket.PREFIX)
                     || ticketId.startsWith(ProxyGrantingTicket.PROXY_GRANTING_TICKET_PREFIX)) {
                 // There is no need to distinguish between TGTs and PGTs since PGTs inherit from TGTs
-                return entityManager.find(TicketGrantingTicketImpl.class, ticketId);
+                return entityManager.find(TicketGrantingTicketImpl.class, ticketId,
+                        lockTgt ? LockModeType.PESSIMISTIC_WRITE : null);
```

Comment:
While this is perfectly adequate, I wonder, would it be more flexible if we allowed ENUMs to be defined as a property? Would adopters need to modify the lock type to something better suited for their deployment? 

I realize this might be overkill. Just thinking out loud here.